routers/web/repo/actions/badge.go

```diff
@@ -19,13 +19,31 @@ import (
 func GetWorkflowBadge(ctx *context.Context) {
 	workflowFile := ctx.PathParam("workflow_name")
 	branch := ctx.Req.URL.Query().Get("branch")
-	if branch == "" {
+	tag := ctx.Req.URL.Query().Get("tag")
+	useLatestTag := ctx.Req.URL.Query().Has("latest_tag")
+	var ref string
+	switch {
+	case useLatestTag:
+		tags, _, err := ctx.Repo.GitRepo.GetTagInfos(0, 1)
+		if err != nil {
+			ctx.ServerError("GetTagInfos", err)
+			return
+		}
+		if len(tags) != 0 {
+			tag = tags[0].Name
+		}
+		ref = fmt.Sprintf("refs/tags/%s", tag)
```

Comment:
Should I fallback to the default branch here if there is no latest tag?

Reply:
No, I think we should return an empty result.